x-pack/plugins/security_solution/public/explore/network/components/embeddables/embedded_map.tsx

```diff
@@ -119,31 +120,55 @@ export const EmbeddedMapComponent = ({
   const { kibanaDataViews } = useDeepEqualSelector((state) => getDataViewsSelector(state));
   const { selectedPatterns } = useSourcererDataView(SourcererScopeName.default);
 
-  const [mapIndexPatterns, setMapIndexPatterns] = useState(
-    kibanaDataViews.filter((dataView) => selectedPatterns.includes(dataView.title))
-  );
+  const isFieldInIndexPattern = useIsFieldInIndexPattern();
+
+  const [mapDataViews, setMapDataViews] = useState<SourcererDataView[]>([]);
+
+  const availableDataViews = useMemo(() => {
+    const dataViews = kibanaDataViews.filter((dataView) =>
+      selectedPatterns.includes(dataView.title)
+    );
+    if (selectedPatterns.length > 0 && dataViews.length === 0) {
+      setIsIndexError(true);
+    }
+    return dataViews;
+  }, [kibanaDataViews, selectedPatterns]);
+
+  useEffect(() => {
+    let canceled = false;
+
+    const fetchData = async () => {
+      try {
+        const apiResponse = await Promise.all(
+          availableDataViews.map(async ({ title }) =>
```

Comment:
We are going to make one extra API call for every data view. Could this significantly impact the network page load time?

Reply:
yes, unfortunately :( 

however, without it we query fields on data views without confirming they actually exist leading to bugs like the screenshot above. it is a bit quicker since we are only asking about certain fields and not requesting all fields. @mattkime added an enhancement for the api to take the new fields argument in order to make this as efficient as possible: https://github.com/elastic/kibana/pull/147666

Reply:
Ok, I think it should be fine. Only the map has to wait for this request.